segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/WriterState.java

```diff
@@ -72,10 +72,9 @@ class WriterState {
     private ArrayList<ErrorContext> errorContexts;
 
     /**
-     * Allowed number of events for index segment append
+     * Expected size of events for index segment append
      */
-    @GuardedBy("this")
-    private final long eventSize;
+    private long eventSize;
```

Comment:
final

Reply:
Making it final gives the compilation error saying 'indexEventSize' might not have been initialized. Since we are calling single arg constructor from two arg constructor, it is not feasible for us to initialize it, So, removed final from it.

Reply:
Not making it final and having two constructors doesn't mean it's not assigned, it's meaning it is being silently assigned to a value, but with no clear representation in the code of what that value is, what it means or what the consequences are. This makes it hard to reason about the code, and it's even more confusing because in a class like this where there is locking, it looks like the field is changing, if it were, then it would need a lock guarding it, so it makes reviewing the code harder because it looks like there is also a thread safety issue. So please explicitly assign a value in the other constructor, and document the meaning of that value so the field can be final.

Reply:
Made the variable as final.